posts/GeoMx-scMask-generation/index.qmd

```diff
@@ -0,0 +1,167 @@
+the usual [caveats](https://nanostring-biostats.github.io/CosMx-Analysis-Scratch-Space/about.html){target="_blank"} and [license](https://nanostring-biostats.github.io/CosMx-Analysis-Scratch-Space/license.html){target="_blank"} applies.
+
+
+
+# Workflow Description
+
+The pipeline is a python script `Generate_scMask_given_query.py` that runs as a command line and you can downloaded from `_code/GeoMx-scMask-generation` [folder](https://github.com/Nanostring-Biostats/CosMx-Analysis-Scratch-Space/tree/Main/_code/GeoMx-scMask-generation){target="_blank"} in this repository. By passing file path to image folder to the script, the pipeline would take GeoMx&#8482; morphology images and generate binary masks for negative-stained cells and cells connecting to positive-stained regions, respectively. 
+
+![Workflow schematic](figures/workflow_GeoMx_scMask_given_query.png){#fig-workflow}
+
+1. The process starts with color-channel reordering of morphology images that user downloads from [GeoMx&#8482; DSP](https://nanostring.com/products/geomx-digital-spatial-profiler/geomx-dsp-overview/){target="_blank"} instrument for the region of interest (ROI). The pipeline script would take those images and rearrange the channel order to have cytoplasm stain as 1st channel, query marker stain as 2nd channel, and optionally nuclear stain as 3rd channel. 
+    + The rearranged images would be saved in `2chan_res` (or `3chan_res` if including nuclear stain as 3rd channel) sub folder of the user-designated output root folder. 
+2. After rearrangement, the pipeline script would perform cell segmentation using cytoplasm stain and optionally nuclear stain as well. The resulting cell label images with pixel value indicates cell ID would be saved under `cellLabels` sub folder within the step (1) result folder.
+3. The pipeline script also takes the query marker stain of each image and performs auto-threshold to identify individual objects with positive stain and **bigger** than user-defined size cutoff, `min_positive_area`.
+4. With label images from both cell segmentation and query-stained object segmentation, the pipeline script would then perform connectivity analysis between the 2 sets of label images and output 2 sets of binary masks in separate folders under user-designated output root folder. 
+    i) `negative_cell_masks`: binary masks for cells that does **NOT** have valid overlapping with any of positive-stained object. 
+         +  A valid overlapping is defined by whether the intersected area between cell and positive-stained object is **bigger** than user-defined size cutoff, `min_intersect_area`. 
+    ii) `positive_ROI_masks`: binary masks for cells with valid overlapping event **AND** their connected positive-stained objects. 
+
+At this point, user can upload the output binary masks back to GeoMx&#8482; DSP instrument for the corresponding ROIs and initiate the sample collection. 
```

Comment:
I think this would be good to put higher up to help explain _why_ a user may want to take this approach.

Reply:
I'm not following. It says in the 1st paragraph that this post is about how to generate GeoMx masks. Could you please clarify how would you like it changed?

Reply:
In the workflow step (3) and (4.i), I added additional bullet points explain how the size cutoff  would help.

step(3)
> You can adjust the size cutoff based on your particular application to get a cleaner isolation of objects of interest. For example, if you are only interested in mature tangles, you can use higher size cutoff to select only the query-positive objects of big size. 

step(4.i)

> The ability to define valid overlapping  based on area cutoff could help to separate between barely touching events and more significant overlapping events. This could be useful if you want to separate between immune cells just contacting a microbe stained in query channel and immune cells that have engulfed an invading microbe.